File: OneDrive 1.2 Feb 23 2024/C++/Simulacion de examen/Publicacion.cpp
// Inclusión de la cabecera de la clase Publicacion y la biblioteca iostream para entrada/salida
#include "Publicacion.h"
#include <iostream>
using namespace std;

// Constructor para inicializar un libro impreso con sus atributos específicos
Publicacion::Publicacion(int pu, string au, string ti, int pa, string cu, string ca, double pr) {
    this->publicacion = pu; // Tipo de publicación
    this->autor = au;       // Autor del libro
    this->titulo = ti;      // Título del libro
    this->paginas = pa;     // Número de páginas del libro
    this->cubierta = cu;    // Tipo de cubierta del libro (pasta dura, pasta blanda)
    this->categoria = ca;   // Categoría a la que pertenece el libro
    this->precio = pr;      // Precio del libro
}

// Constructor para inicializar un audiolibro con sus atributos específicos
Publicacion::Publicacion(int pu, string au, string ti, int tr, string ca, double pr) {
    this->publicacion = pu; // Tipo de publicación
    this->autor = au;       // Autor del audiolibro
    this->titulo = ti;      // Título del audiolibro
    this->tracks = tr;      // Número de tracks del audiolibro
    this->categoria = ca;   // Categoría a la que pertenece el audiolibro
    this->precio = pr;      // Precio del audiolibro
}

// Constructor para inicializar una revista con sus atributos específicos
Publicacion::Publicacion(int pu, string ti, int nu, int vo, string ca, double pr) {
    this->publicacion = pu; // Tipo de publicación
    this->titulo = ti;      // Título de la revista
    this->numero = nu;      // Número de la revista
    this->volumen = vo;     // Volumen de la revista
    this->categoria = ca;   // Categoría a la que pertenece la revista
    this->precio = pr;      // Precio de la revista
}

// Métodos getter para acceder a los atributos de la clase
int Publicacion::getPublicacion() { return this->publicacion; }
string Publicacion::getAutor() { return this->autor; }
string Publicacion::getTitulo() { return this->titulo; }
int Publicacion::getPaginas() { return this->paginas; }
string Publicacion::getCubierta() { return this->cubierta; }
int Publicacion::getTracks() { return this->tracks; }
string Publicacion::getCategoria() { return this->categoria; }
int Publicacion::getNumero() { return this->numero; }
int Publicacion::getVolumen() { return this->volumen; }
double Publicacion::getPrecio() { return this->precio; }

// Métodos setter para modificar los atributos de la clase
void Publicacion::setPublicacion(int x) { this->publicacion = x; }
void Publicacion::setAutor(string x) { this->autor = x; }
void Publicacion::setTitulo(string x) { this->titulo = x; }
void Publicacion::setPaginas(int x) { this->paginas = x; }
void Publicacion::setCubierta(string x) { this->cubierta = x; }
void Publicacion::setTracks(int x) { this->tracks = x; }
void Publicacion::setCategoria(string x) { this->categoria = x; }
void Publicacion::setNumero(int x) { this->numero = x; }
void Publicacion::setVolumen(int x) { this->volumen = x; }
void Publicacion::setPrecio(double x) { this->precio = x; }

// Método para imprimir la información de la publicación basada en su tipo
void Publicacion::Imprimir() {
    switch (this->publicacion) {
    case 1: // Libro impreso
        cout << "Tipo de publicacion: Libro impreso" << endl;
        cout << "Datos de autor: " << this->getAutor() << endl;
        cout << "Titulo: " << this->getTitulo() << endl;
        cout << "Número de páginas: " << getPaginas() << endl;
        cout << "Tipo de cubierta: " << this->getCubierta() << endl;
        cout << "Categoría: " << this->getCategoria() << endl;
        cout << "Precio: " << this->getPrecio() << endl;
        break;
    case 2: // Audio Libro
        cout << "Tipo de publicacion: Audio Libro" << endl;
        cout << "Datos de autor: " << this->getAutor() << endl;
        cout << "Titulo: " << this->getTitulo() << endl;
        cout << "Número de tracks: " << this->getTracks() << endl;
        cout << "Categoría: " << this->getCategoria() << endl;
        cout << "Precio: " << this->getPrecio() << endl;
        break;
    case 3: // Revista
        cout << "Tipo de publicacion: Revista" << endl;
        cout << "Titulo: " << this->getTitulo() << endl;
        cout << "Número de revista: " << this->getNumero() << endl;
        cout << "Volumen: " << this->getVolumen() << endl;
        cout << "Categoría: " << this->getCategoria() << endl;
        cout << "Precio: " << this->getPrecio() << endl;
        break;
    }
}

File: OneDrive 1.2 Feb 23 2024/Java/Simulacion de examen/Publicacion.java
// Definición de la clase Publicacion
public class Publicacion {
    // Atributos privados de la clase para almacenar características de una publicación
    private int publicacion; // Tipo de publicación (1: Libro impreso, 2: Audio libro, 3: Revista)
    private String autor; // Nombre del autor de la publicación
    private String titulo; // Título de la publicación
    private int paginas; // Número de páginas (aplica para libros y revistas)
    private String cubierta; // Tipo de cubierta para libros (pasta dura, pasta blanda)
    private int tracks; // Número de tracks (aplica para audiolibros)
    private String categoria; // Categoría de la publicación (literatura, consulta, etc.)
    private int numero; // Número de revista (aplica para revistas)
    private int volumen; // Volumen de la revista (aplica para revistas)
    private double precio; // Precio de la publicación

    // Constructor para Libro Impreso
    public Publicacion(int pu, String au, String ti, int pa, String cu, String ca, double pr){
        this.publicacion = pu;
        this.autor = au;
        this.titulo = ti;
        this.paginas = pa;
        this.cubierta = cu;
        this.categoria = ca;
        this.precio = pr;
    }

    // Constructor para Audio Libro
    public Publicacion(int pu, String au, String ti, int tr, String ca, double pr){
        this.publicacion = pu;
        this.autor = au;
        this.titulo = ti;
        this.tracks = tr;
        this.categoria = ca;
        this.precio = pr;
    }

    // Constructor para Revista
    public Publicacion(int pu, String ti, int nu, int vo, String ca, double pr){
        this.publicacion = pu;
        this.titulo = ti;
        this.numero = nu;
        this.volumen = vo;
        this.categoria = ca;
        this.precio = pr;
    }

    // Métodos getter para obtener los valores de los atributos
    public int getPublicacion(){
       return this.publicacion;
    }

    public String getAutor(){
        return this.autor;
    }

    public String getTitulo(){        
        return this.titulo;
    }

    public int getPaginas(){
        return this.paginas;
    }

    public String getCubierta(){
        return this.cubierta;
    }

    public int getTracks(){
        return this.tracks;
    }

    public String getCategoria(){
        return this.categoria;
    }

    public int getNumero(){
        return this.numero;
    }

    public int getVolumen(){
        return this.volumen;
    }

    public double getPrecio(){
        return this.precio;
    }

    // Métodos setter para establecer los valores de los atributos
    void setPublicacion(int x){
        this.publicacion = x;
    }

    void setAutor(String x){
        this.autor = x;
    }

    void setTitulo(String x){        
        this.titulo = x;
    }

    void setPaginas(int x){
        this.paginas = x;
    }

    void setCubierta(String x){
        this.cubierta = x;
    }

    void setTracks(int x){
        this.tracks = x;
    }

    void setCategoria(String x){
        this.categoria = x;
    }

    void setNumero(int x){
        this.numero = x;
    }

    void setVolumen(int x){
        this.volumen = x;
    }

    void setPrecio(double x){
        this.precio = x;
    }

    // Método para imprimir los detalles de la publicación basado en su tipo
    void Imprimir(){
        switch(this.publicacion){
            case 1: // Libro impreso
                System.out.println("Tipo de publicacion: Libro impreso");
                System.out.println("Datos de autor: "+this.getAutor());
                System.out.println("Titulo: "+this.getTitulo());
                System.out.println("Número de páginas: "+getPaginas());
                System.out.println("Tipo de cubierta: "+this.getCubierta());
                System.out.println("Categoría: "+ this.getCategoria());
                System.out.println("Precio: "+this.getPrecio());
                break;
            case 2: // Audio libro
                System.out.println("Tipo de publicacion: Audio Libro");
                System.out.println("Datos de autor: "+this.getAutor());
                System.out.println("Titulo: "+this.getTitulo());
                System.out.println("Número de tracks: "+getTracks());
                System.out.println("Categoría: "+ this.getCategoria());
                System.out.println("Precio: "+this.getPrecio());
                break;
            case 3: // Revista
                System.out.println("Tipo de publicacion: Revista");
                System.out.println("Titulo: "+this.getTitulo());
                System.out.println("Número de revista: "+getNumero());
                System.out.println("Volumen: "+this.getVolumen());
                System.out.println("Categoría: "+ this.getCategoria());
                System.out.println("Precio: "+this.getPrecio());
                break;
        }
    }
}
